cmd/fluxctl/setup_cmd.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/weaveworks/flux/git"
+)
+
+type setupOpts struct {
+	gitURL             string
+	gitBranch          string
+	gitFluxPath        string
+	gitLabel           string
+	timeout            time.Duration
+	namespace          string
+	additionalFluxArgs []string
+	amend              bool
+}
+
+func newSetup() *setupOpts {
+	return &setupOpts{}
+}
+
+func (opts *setupOpts) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "setup",
+		Short: "Bootstrap Flux, installing it in the cluster and initializing its manifests in the Git repository",
+		RunE:  opts.RunE,
+	}
+	cmd.Flags().StringVarP(&opts.gitURL, "git-url", "r", "",
+		"URL of the Git repository to be used by Flux, e.g. git@github.com:<your username>/flux-get-started")
+	cmd.Flags().StringVarP(&opts.gitBranch, "git-branch", "b", "master",
+		"Git branch to be used by Flux")
+	cmd.Flags().StringVarP(&opts.gitLabel, "git-label", "l", "flux",
+		"Git label to keep track of FLux's sync progress; overrides both --git-sync-tag and --git-notes-ref")
+	cmd.Flags().StringVarP(&opts.gitFluxPath, "git-flux-subdir", "p", "flux/",
+		"Directory within the Git repository where to commit the Flux manifests")
+	cmd.Flags().DurationVarP(&opts.timeout, "timeout", "t", 10*time.Second,
+		"Timeout duration for I/O operations")
+	cmd.Flags().StringVarP(&opts.namespace, "namespace", "n", "flux",
+		"Cluster namespace where to install flux")
+	cmd.Flags().BoolVarP(&opts.amend, "amend", "a", false,
+		"Stop for amending the Flux manifests before pushing them to the Git repository")
+	cmd.Flags().StringSliceVarP(&opts.additionalFluxArgs, "extra-flux-args", "e", []string{},
+		"Additional arguments for Flux as CSVs, e.g. --extra-flux-arg='--manifest-generation=true,--sync-garbage-collection=true'")
+	return cmd
+}
+
+func (opts *setupOpts) RunE(cmd *cobra.Command, args []string) error {
+	//0. Read and patch embedded deploy/* manifests with what was passed to ops
+
+	// TODO. BTW, it's probably easier to embed chart/flux/* and use `helm template` as a library instead
```

Comment:
That's the same approach used by`linkerd install`: https://github.com/linkerd/linkerd2/blob/edge-19.7.4/cli/cmd/install.go#L745